Yep, the `pkgsCross.riscv64.bpftools` fails with these patches ():
```diff
diff --git a/pkgs/by-name/bp/bpftools/package.nix b/pkgs/by-name/bp/bpftools/package.nix
index 88ec301bc356..2f2de8e6aaf4 100644
--- a/pkgs/by-name/bp/bpftools/package.nix
+++ b/pkgs/by-name/bp/bpftools/package.nix
@@ -1,6 +1,7 @@
 {
   lib,
   stdenv,
+  fetchpatch,
   linuxHeaders,
   buildPackages,
   libopcodes,
@@ -25,6 +26,25 @@ stdenv.mkDerivation rec {
   patches = [
     # fix unknown type name '__vector128' on ppc64le
     ./include-asm-types-for-ppc64le.patch
+    # fix build for riscv64
+    (fetchpatch {
+      # libbpf: Add missing per-arch include path
+      # https://patchwork.kernel.org/project/linux-riscv/patch/20240927131355.350918-1-bjorn@kernel.org/
+      url = "https://patchwork.kernel.org/project/linux-riscv/patch/20240927131355.350918-1-bjorn@kernel.org/raw/";
+      hash = "sha256-edXY/ejHW5L7rGgY5B2GmVZxUgnLdBadNhBOSAgcL7M=";
+    })
+    (fetchpatch {
+      # selftests: bpf: Add missing per-arch include path
+      # https://patchwork.kernel.org/project/linux-riscv/patch/20240927131355.350918-2-bjorn@kernel.org/
+      url = "https://patchwork.kernel.org/project/linux-riscv/patch/20240927131355.350918-2-bjorn@kernel.org/raw/";
+      hash = "sha256-7yNWE/L/qd3vcLtJYoSyGxB3ytySlr20R0D3t5ni2Fc=";
+    })
+    (fetchpatch {
+      # tools: Override makefile ARCH variable if defined, but empty
+      # https://patchwork.kernel.org/project/linux-riscv/patch/20241127101748.165693-1-bjorn@kernel.org/
+      url = "https://patchwork.kernel.org/project/linux-riscv/patch/20241127101748.165693-1-bjorn@kernel.org/raw/";
+      hash = "sha256-y8N71Hm1XfX9g3S6PzW2m7Lxp6wQQMfQE9L0QNt8cYY=";
+    })
   ];

   depsBuildBuild = [ buildPackages.stdenv.cc ];
```

```
this derivation will be built:
  /nix/store/jv7a36zrl1qmm5h2d0vvy8ykg9qvx7q2-bpftools-riscv64-unknown-linux-gnu-6.12.drv
building '/nix/store/jv7a36zrl1qmm5h2d0vvy8ykg9qvx7q2-bpftools-riscv64-unknown-linux-gnu-6.12.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/r38bz8v56dxsylmzchi61zvm63nbn42s-linux-6.12.tar.xz
source root is linux-6.12
setting SOURCE_DATE_EPOCH to timestamp 1731881708 of file "linux-6.12/virt/lib/irqbypass.c"
Running phase: patchPhase
applying patch /nix/store/64hqs37pqw2lrijgi2v3rq8dv48nfrcv-include-asm-types-for-ppc64le.patch
patching file tools/include/uapi/linux/types.h
applying patch /nix/store/ykmsq63zdbxsd55l4cn88934hbv5dvrq-raw
patching file tools/lib/bpf/Makefile
applying patch /nix/store/k3fafs82wzfbb4jn87l95430ywxrxlvs-raw
patching file tools/testing/selftests/bpf/Makefile
applying patch /nix/store/l8pnq31ljplxzzl4l7l65ygrgpibcp6h-raw
patching file tools/scripts/Makefile.arch
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
patching script interpreter paths in scripts/bpf_doc.py
scripts/bpf_doc.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/dbs6x9wv3q9mf90s2fx88bhwipvqx608-python3-3.12.8/bin/python3"
substituteStream() in derivation bpftools-riscv64-unknown-linux-gnu-6.12: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file './bpftool/Makefile')
no configure script, doing nothing
Running phase: buildPhase
build flags: SHELL=/nix/store/66902k906fc5p0xdjx7kbz0r00lgm6zy-bash-5.2p37/bin/bash bpftool bpf_asm bpf_dbg

Auto-detecting system features:
...                                  libbfd: [ on  ]

  DESCEND bpftool

Auto-detecting system features:
...                         clang-bpf-co-re: [ OFF ]
...                                    llvm: [ OFF ]
...                                  libcap: [ OFF ]
...                                  libbfd: [ on  ]

  MKDIR   /build/linux-6.12/tools/bpf/bpftool/libbpf/
  GEN     /build/linux-6.12/tools/bpf/bpftool/libbpf/bpf_helper_defs.h
  CC      /build/linux-6.12/tools/bpf/bpftool/libbpf/staticobjs/libbpf.o
In file included from /build/linux-6.12/tools/include/asm/barrier.h:12,
                 from /build/linux-6.12/tools/include/linux/ring_buffer.h:4,
                 from libbpf.c:37:
/build/linux-6.12/tools/include/asm/../../arch/riscv/include/asm/barrier.h:13:10: fatal error: asm/fence.h: No such file or directory
   13 | #include <asm/fence.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[3]: *** [/build/linux-6.12/tools/build/Makefile.build:106: /build/linux-6.12/tools/bpf/bpftool/libbpf/staticobjs/libbpf.o] Error 1
make[2]: *** [Makefile:165: /build/linux-6.12/tools/bpf/bpftool/libbpf/staticobjs/libbpf-in.o] Error 2
make[1]: *** [Makefile:46: /build/linux-6.12/tools/bpf/bpftool/libbpf/libbpf.a] Error 2
make: *** [Makefile:113: bpftool] Error 2
```

--- a/pkgs/by-name/bp/bpftools/package.nix
+++ b/pkgs/by-name/bp/bpftools/package.nix
@@ -1,6 +1,7 @@
 {
   lib,
   stdenv,
+  fetchpatch,
   linuxHeaders,
   buildPackages,
   libopcodes,
@@ -25,6 +26,25 @@ stdenv.mkDerivation rec {
   patches = [
     # fix unknown type name '__vector128' on ppc64le
     ./include-asm-types-for-ppc64le.patch
+    # fix build for riscv64
+    (fetchpatch {
+      # libbpf: Add missing per-arch include path
+      # https://patchwork.kernel.org/project/linux-riscv/patch/20240927131355.350918-1-bjorn@kernel.org/
+      url = "https://patchwork.kernel.org/project/linux-riscv/patch/20240927131355.350918-1-bjorn@kernel.org/raw/";
+      hash = "sha256-edXY/ejHW5L7rGgY5B2GmVZxUgnLdBadNhBOSAgcL7M=";
+    })
+    (fetchpatch {
+      # selftests: bpf: Add missing per-arch include path
+      # https://patchwork.kernel.org/project/linux-riscv/patch/20240927131355.350918-2-bjorn@kernel.org/
+      url = "https://patchwork.kernel.org/project/linux-riscv/patch/20240927131355.350918-2-bjorn@kernel.org/raw/";
+      hash = "sha256-7yNWE/L/qd3vcLtJYoSyGxB3ytySlr20R0D3t5ni2Fc=";
+    })
+    (fetchpatch {
+      # tools: Override makefile ARCH variable if defined, but empty
+      # https://patchwork.kernel.org/project/linux-riscv/patch/20241127101748.165693-1-bjorn@kernel.org/
+      url = "https://patchwork.kernel.org/project/linux-riscv/patch/20241127101748.165693-1-bjorn@kernel.org/raw/";
+      hash = "sha256-y8N71Hm1XfX9g3S6PzW2m7Lxp6wQQMfQE9L0QNt8cYY=";
+    })
   ];
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];
@@ -67,6 +87,9 @@ stdenv.mkDerivation rec {
     "bpf_asm"
     "bpf_dbg"
   ];
+
+  # needed for cross to riscv64
+  makeFlags = [ "ARCH=${stdenv.hostPlatform.linuxArch}" ];
 
   installPhase = ''
     make -C bpftool install

